drivers/ma_yya_report/extensions/grda_warehouse/hmis_form_extension.rb

```diff
@@ -0,0 +1,88 @@
+module MaYyaReport
+  module GrdaWarehouse
+    module HmisFormExtension
+      extend ActiveSupport::Concern
+
+      def current_school_attendance
+        # First look for the HUD questions
+        answers[:sections].each do |s|
+          s[:questions].each do |q|
+            return HUD.current_school_attended(q[:answer], true) if q[:answer].present? && q[:question] == 'Current school enrollment and attendance'
+          end
+        end
+
+        # If we didn't find one, try the older format
+        answers[:sections].each do |s|
+          s[:questions].each do |q|
+            response = q[:answer] if q[:answer].present? && q[:question] == 'Level of education'
+            return 2 if response.contains?('YYA is currently enrolled in college|') # Assume attending if enrolled
+            return 0 if response.contains?('YYA has completed high school or GED/HISET|') # If completed, but not in college, assume not enrolled
+          end
+        end
+
+        # Finally, give up
+        nil
+      end
+
+      def most_recent_educational_status
+        # First look for the HUD questions
+        answers[:sections].each do |s|
+          s[:questions].each do |q|
+            return HUD.most_recent_ed_status(q[:answer], true) if q[:answer].present? && q[:question] == 'C3.A'
+          end
```

Comment:
I think you want `HmisForm.answer_from_section(section, question_string)`  Though, if you don't know the section you expect it in, that could be slightly less efficient.